salt/top.sls

```diff
@@ -10,10 +10,19 @@ base:
     - rebootmgr
     - transactional-update
     - haproxy
+    - kubectl-config
   'roles:admin':
     - match: grain
     - velum
     - ldap
+  'roles:etcd':
+    - match: grain
+    - etcd
+  'roles:kube-master':
```

Comment:
Why the change of order of this specific bit?

Reply:
The initial starting point for salt as it was ordering the states was wrong, I believe the requires alone are probably enough to resolve this, but, still - it's better to put the "starting order" salt uses to seed the dependency graph into the rough order we expect things to be deployed.

Reply:
It's tricky because visually a kube-master on the `top.sls` looks like this:

```
base:
  'roles:(admin|kube-(master|minion))':
    - match: grain_pcre
    - swap
    - etc-hosts
    - proxy
    - rebootmgr
    - transactional-update
    - haproxy
    - kubectl-config
  'roles:etcd': (optional, depends if it has the `etcd` role too)
    - match: grain
    - etcd
  'roles:kube-master':
    - match: grain
    - kube-apiserver
    - kube-controller-manager
    - kube-scheduler
  'roles:kube-(master|minion)':
    - match: grain_pcre
    - ca-cert
    - repositories
    - motd
    - users
    - cert
    - docker
    - container-feeder
    - kubelet
    - kube-proxy
```

So visually we are also applying `kube-apiserver` before `ca-cert`, for example... but anyways, as you say this is not an enforcement of order. In fact, ordering semantics in salt are... "special".